homeassistant/helpers/entity_platform.py

```diff
@@ -518,33 +518,55 @@ async def async_add_entities(
         hass = self.hass
 
         entity_registry = ent_reg.async_get(hass)
-        tasks = [
+        coros = [
             self._async_add_entity(entity, update_before_add, entity_registry)
             for entity in new_entities
         ]
 
         # No entities for processing
-        if not tasks:
+        if not coros:
             return
 
-        timeout = max(SLOW_ADD_ENTITY_MAX_WAIT * len(tasks), SLOW_ADD_MIN_TIMEOUT)
-        try:
-            async with self.hass.timeout.async_timeout(timeout, self.domain):
-                await asyncio.gather(*tasks)
-        except TimeoutError:
-            self.logger.warning(
-                "Timed out adding entities for domain %s with platform %s after %ds",
-                self.domain,
-                self.platform_name,
-                timeout,
-            )
-        except Exception:
-            self.logger.exception(
-                "Error adding entities for domain %s with platform %s",
-                self.domain,
-                self.platform_name,
-            )
-            raise
+        timeout = max(SLOW_ADD_ENTITY_MAX_WAIT * len(coros), SLOW_ADD_MIN_TIMEOUT)
+        async with self.hass.timeout.async_timeout(timeout, self.domain):
+            if update_before_add:
+                try:
+                    await asyncio.gather(*coros)
```

Comment:
This one is a bit of a unexpected case because the behavior here when something raises means we continue on and the adds continue to run in the background.

Reply:
I added return_exceptions so we don't stop waiting for the other entities if one fails